ledger/accountdb.go

```diff
@@ -1026,74 +1080,395 @@ func accountsPutTotals(tx *sql.Tx, totals ledgercore.AccountTotals, catchpointSt
 	return err
 }
 
-// accountsNewRound updates the accountbase and assetcreators tables by applying the provided deltas to the accounts / creatables.
-// The function returns a dbAccountData for the modified accounts which can be stored in the base cache.
-func accountsNewRound(tx *sql.Tx, updates compactAccountDeltas, creatables map[basics.CreatableIndex]ledgercore.ModifiedCreatable, proto config.ConsensusParams, lastUpdateRound basics.Round) (updatedAccounts []dbAccountData, err error) {
+func loadHoldings(stmt *sql.Stmt, eah ledgercore.ExtendedAssetHolding) (map[basics.AssetIndex]basics.AssetHolding, ledgercore.ExtendedAssetHolding, error) {
```

Comment:
you're calling this method from two places, and in both you're ignoring the returned `ledgercore.ExtendedAssetHolding`. If this is an input/output parameter, maybe you should pass it via a pointer ? I think that the fact you're doing it by copy might be an issue here since you're changing an internal array element, which would change the source, but then - you're not updating the source correctly.

Reply:
I'm still worried about the usage of this method.  It's creating a *copy* and making changes that would propagate to the origin. It's only valid when used in the following "pattern":
```go
_, extholdings = loadHoldings(stmt, extholdings)
```
if you were to use the following, it would be broken:
```go
_, newholding = loadHoldings(stmt, extholdings)
```
since the `extholdings` would included the loaded groups, but not the loaded flag.

Reply:
yes you right. I fixed it by writing groups into a temp array.